flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java

```diff
@@ -226,6 +226,53 @@ public void shouldUpdateExistingLocalFormPackageToNpmPackage()
                 .getObject(NodeUpdater.DEPENDENCIES).getString(formPackage));
     }
 
+    @Test
+    public void shouldSkipUpdatingNonParsableVersions() throws IOException {
+        JsonObject packageJson = Json.createObject();
+        JsonObject dependencies = Json.createObject();
+        packageJson.put(NodeUpdater.DEPENDENCIES, dependencies);
+        JsonObject vaadinDependencies = Json.createObject();
+        vaadinDependencies.put(NodeUpdater.DEPENDENCIES, Json.createObject());
+        packageJson.put(NodeUpdater.VAADIN_DEP_KEY, vaadinDependencies);
+
+        String formPackage = "@vaadin/form";
+        String existingVersion = "../../../some/local/path";
+        String newVersion = "2.0.0";
+
+        dependencies.put(formPackage, existingVersion);
+
+        nodeUpdater.addDependency(packageJson, NodeUpdater.DEPENDENCIES,
+                formPackage, newVersion);
+
+        Assert.assertEquals(existingVersion, packageJson
+                .getObject(NodeUpdater.DEPENDENCIES).getString(formPackage));
+    }
+
+    @Test
+    public void shouldThrowExceptionOnNewVersionNonParsable() {
+        JsonObject packageJson = Json.createObject();
+        JsonObject dependencies = Json.createObject();
+        packageJson.put(NodeUpdater.DEPENDENCIES, dependencies);
+        JsonObject vaadinDependencies = Json.createObject();
+        vaadinDependencies.put(NodeUpdater.DEPENDENCIES, Json.createObject());
+        packageJson.put(NodeUpdater.VAADIN_DEP_KEY, vaadinDependencies);
+
+        String formPackage = "@vaadin/form";
+        String existingVersion = "2.0.0";
+        String newVersion = "../../../some/local/path";
+
+        dependencies.put(formPackage, existingVersion);
+
+        try {
+            nodeUpdater.addDependency(packageJson, NodeUpdater.DEPENDENCIES,
+                    formPackage, newVersion);
+            Assert.fail("Expected NumberFormatException");
+        } catch (NumberFormatException e) {
+            Assert.assertTrue(
+                    e.getMessage().contains("is not a valid version"));
+        }
```

Comment:
nit: try/catch can be replaced with `Assert.assertThrows`
```suggestion
        NumberFormatException expectedException = Assert
                .assertThrows(NumberFormatException.class,
                        () -> nodeUpdater.addDependency(packageJson,
                                NodeUpdater.DEPENDENCIES, formPackage,
                                newVersion));
        Assert.assertTrue(expectedException.getMessage()
                .contains("is not a valid version"));
```